Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -646,6 +647,207 @@ void orient_to_bound_a_volume(TriangleMesh& tm)
 {
   orient_to_bound_a_volume(tm, parameters::all_default());
 }
+
+/*!
+ * \ingroup PMP_orientation_grp
+ * reverses the connected components of `tm` having compatible boundary cycles
+ * that could be merged if their orientation were made compatible, and stitches them.
+ * Connected components are examined by increasing number of faces.
+ *
+ * @tparam PolygonMesh a model of `MutableFaceGraph`, `HalfedgeListGraph` and `FaceListGraph`.
+ * @tparam NamedParameters a sequence of \ref pmp_namedparameters
+ *
+ * @param pm a surface mesh
+ * @param np optional sequence of \ref pmp_namedparameters among the ones listed below
+ *
+ * \cgalNamedParamsBegin
+ *   \cgalParamBegin{vertex_point_map}
+ *     the property map with the points associated to the vertices of `pm`.
+ *     If this parameter is omitted, an internal property map for
+ *     `CGAL::vertex_point_t` must be available in `PolygonMesh`
+ *   \cgalParamEnd
+ *   \cgalParamBegin{face_index_map}
+ *     a property map containing an index for each face initialized from 0 to num_faces(pm).
+ *   \cgalParamEnd
+ *   \cgalParamBegin{maximum_number_of_faces}
+ *     if not 0 (default), a connected component is considered reversible only
+ *     if it has no more faces than the value given. Otherwise, it is always considered reversible.
+ *   \cgalParamEnd
+ * \cgalNamedParamsEnd
+ */
+template <class PolygonMesh, class NamedParameters>
+void merge_reversible_connected_components(PolygonMesh& pm,
+                                           const NamedParameters& np)
+{
+  typedef boost::graph_traits<PolygonMesh> GrT;
+  typedef typename GrT::face_descriptor face_descriptor;
+  typedef typename GrT::halfedge_descriptor halfedge_descriptor;
+
+  typedef typename GetVertexPointMap<PolygonMesh, NamedParameters>::const_type Vpm;
+
+  typedef typename boost::property_traits<Vpm>::value_type Point_3;
+  Vpm vpm = parameters::choose_parameter(parameters::get_parameter(np, internal_np::vertex_point),
+                                         get_const_property_map(vertex_point, pm));
+
+  typedef std::size_t F_cc_id;
+  typedef std::size_t B_cc_id;
+
+  typedef typename GetFaceIndexMap<PolygonMesh, NamedParameters>::type Fidmap;
```

Comment:
Actually the branch must be updated to use the new initialized index map API too.